system/HTTP/CURLRequest.php

```diff
@@ -129,6 +135,20 @@ public function __construct(App $config, URI $uri, ?ResponseInterface $response
 
         $this->config = $this->defaultConfig;
         $this->parseOptions($options);
+
+        // Share Connection
+        $optShareConnection = config(ConfigCURLRequest::class)->shareConnectionOptions ?? [
+            CURL_LOCK_DATA_CONNECT,
+            CURL_LOCK_DATA_DNS,
+        ];
+
+        if ($optShareConnection !== []) {
+            $this->shareConnection = curl_share_init();
+
+            foreach (array_unique($optShareConnection) as $opt) {
```

Comment:
Is the array_unique needed? If yes, please add a test to cover this.

Reply:
When i see persistent logic in PHP 8.5, automatically remove duplicate

Reply:
I cant applied test to see content object `CURLOPT_SHARE`

Reply:
> When i see persistent logic in PHP 8.5, automatically remove duplicate

sorry, I did not understand you here. do you mean PHP 8.5 automatically remove duplicates?

Reply:
Yes automatically remove duplicates

Reply:
Ok. No need then.